packages/container/libraries/strongly-typed/README.md

```diff
@@ -108,3 +146,39 @@ const parent = new TypedContainer<{foo: Foo}>();
 const child = parent.createChild<{foo: Bar}>();
 const resolved: Bar = child.get('foo');
 ```
+
+
+## Known issues / limitations
+
+### Strongly-typing `private` properties
+
+Note that because the decorator types can't "see" `private` properties, we can't strongly type them:
+
+```ts
+@injectable()
+class A {
+  @$inject('foo')
+  private foo: Foo; // fails :'(
+}
+```
+
+Work around this by either:
+
+  1. Making the property `public` (and perhaps prefixing it with an underscore to remind consumers it should be private)
+  2. Using the weakly-typed `@inject()` directly from `'inversify'`
+
+
+### Confusing compiler errors for constructor injection
+
+There's no custom compiler errors for TypeScript, so we can't signal particularly usefully that a constructor parameter is wrong, apart from the fact that the compilation fails.
+
+The error will look something like:
+
+```
+Unable to resolve signature of parameter decorator when called as an expression.
+  Argument of type '2' is not assignable to parameter of type 'undefined'. (ts1239)
```

Comment:
@notaphplover I couldn't really get a very useful compiler message either. I went with this, which is a bit different to your proposal, because:

- the type definition was simpler
- the error message at least flags that there's something wrong with argument `2`, which is sort of meaningful

I don't think there's much we can do without this: https://github.com/microsoft/TypeScript/issues/23689

Reply:
Agree, the error is way better than the one experienced with the approach I suggested.

Regarding the ts limitation, I'm afraid so.